app/bundles/FormBundle/Tests/Controller/FormControllerFunctionalTest.php

```diff
@@ -195,6 +210,218 @@ public function testMappedFieldIsNotAutoFilledWhenUpdatingField(): void
         $this->assertStringContainsString('<option value="email">Email</option>', $options, 'Email option should not be pre-selected.');
     }
 
+    /**
+     * @param array{
+     *      type: string,
+     *      properties: array<string, mixed>,
+     *      entities?: array<object>
+     *  } $inputValues The input configuration for the form action
+     * @param array<int, array{
+     *      message: string,
+     *      message_arg: array<string, mixed>
+     *  }> $expectedMessages The expected messages with translation arguments
+     *
+     * @dataProvider dataTestLabelsForFormActions
+     */
+    public function testLabelsForFormAction(array $inputValues, array $expectedMessages): void
+    {
+        $form = $this->createForm('test', 'test');
+
+        // Persist entities if provided
+        if (!empty($inputValues['entities'])) {
+            // @phpstan-ignore-next-line
+            array_map([$this->em, 'persist'], $inputValues['entities']);
+        }
+
+        // create form action
+        $action = $this->createFormAction($form, $inputValues['type'], $inputValues['properties']);
+        $form->addAction(0, $action);
+        $this->em->persist($form);
+
+        $this->em->flush();
+        $this->em->clear();
+
+        $crawler = $this->client->request('GET', sprintf('/s/forms/edit/%d', $form->getId()));
+
+        /** @var TranslatorInterface $translator */
+        $translator = $this->getContainer()->get('translator');
```

Comment:
Assert is better than the comment.
```suggestion
        $translator = $this->getContainer()->get('translator');
        \assert($translator instanceof TranslatorInterface);
```